src/Common/src/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -305,6 +313,11 @@ private static ComputedInstanceFieldLayout ComputeExplicitFieldLayout(MetadataTy
             int packingSize = ComputePackingSize(type, layoutMetadata);
             LayoutInt largestAlignmentRequired = LayoutInt.One;
 
+            if (type.IsObject)
+            {
+                PrepareRuntimeSpecificSystemObjectLayout(type.Context, ref cumulativeInstanceFieldPos, ref largestAlignmentRequired);
```

Comment:
There may be interesting interplay between this and my current statics placement change.

Reply:
I'm replacing with Michal's suggestion to use a plug in type layout algorithm so hopefully not now.